### ISSUE TITLE
Invalid signature on proposed block

### ISSUE THREAD

**Issue by ajsutton** [Consensys-Incorporated/teku#2016]
### Description
Found an error in the logs indicating a block we proposed had an invalid signature.  Need to understand how that's possible then fix it.

```
 tech.pegasys.teku.core.StateTransitionException: tech.pegasys.teku.core.exceptions.BlockProcessingException: tech.pegasys.teku.core.exceptions.BlockProcessingException: Invalid block signature: SignedBeaconBlock{message=BeaconBlock{root=0xc9092218ef597dd409da906d15e405bd7047c8784d5ba3885c2659dfc0c7ed20, slot=38522, proposer_index=538, parent_root=0xdd10c4319dc058074d789e2162acee3b937c0eaf6c021171732f89a203b0191a, state_root=0x4a6bd0705f9bc5c57b0d5414249d9ee7fade0088fd1655eb4ba9fe31f5f60d3c, body=0xa9cde75aea4c89281900e8f5021cebbcbbeeb131e4059aa277ffa09dddfa1d5d}, signature=0xb7d8908679ce6163a64e956cdec9529f6744bd14da09935e4cca17362cc07424a4db3d56b3aa21500ba33bfbb3dc1c7c185d13d9b35ed927957d6e377189bb1c2dc3f45d0afd4bcacdb344a869b7eb58d497e84284ae8264daf779d84f1d5bb3}
	at tech.pegasys.teku.core.StateTransition.initiate(StateTransition.java:112)
	at tech.pegasys.teku.core.ForkChoiceUtil.on_block(ForkChoiceUtil.java:175)
	at tech.pegasys.teku.statetransition.forkchoice.ForkChoice.onBlock(ForkChoice.java:63)
	at tech.pegasys.teku.statetransition.blockimport.BlockImporter.importBlock(BlockImporter.java:56)
	at tech.pegasys.teku.statetransition.blockimport.BlockImporter.onBlockProposed(BlockImporter.java:85)
	at jdk.internal.reflect.GeneratedMethodAccessor79.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: tech.pegasys.teku.core.exceptions.BlockProcessingException: tech.pegasys.teku.core.exceptions.BlockProcessingException: Invalid block signature: SignedBeaconBlock{message=BeaconBlock{root=0xc9092218ef597dd409da906d15e405bd7047c8784d5ba3885c2659dfc0c7ed20, slot=38522, proposer_index=538, parent_root=0xdd10c4319dc058074d789e2162acee3b937c0eaf6c021171732f89a203b0191a, state_root=0x4a6bd0705f9bc5c57b0d5414249d9ee7fade0088fd1655eb4ba9fe31f5f60d3c, body=0xa9cde75aea4c89281900e8f5021cebbcbbeeb131e4059aa277ffa09dddfa1d5d}, signature=0xb7d8908679ce6163a64e956cdec9529f6744bd14da09935e4cca17362cc07424a4db3d56b3aa21500ba33bfbb3dc1c7c185d13d9b35ed927957d6e377189bb1c2dc3f45d0afd4bcacdb344a869b7eb58d497e84284ae8264daf779d84f1d5bb3}
	at tech.pegasys.teku.core.StateTransition.initiate(StateTransition.java:103)
	... 13 more
Caused by: tech.pegasys.teku.core.exceptions.BlockProcessingException: Invalid block signature: SignedBeaconBlock{message=BeaconBlock{root=0xc9092218ef597dd409da906d15e405bd7047c8784d5ba3885c2659dfc0c7ed20, slot=38522, proposer_index=538, parent_root=0xdd10c4319dc058074d789e2162acee3b937c0eaf6c021171732f89a203b0191a, state_root=0x4a6bd0705f9bc5c57b0d5414249d9ee7fade0088fd1655eb4ba9fe31f5f60d3c, body=0xa9cde75aea4c89281900e8f5021cebbcbbeeb131e4059aa277ffa09dddfa1d5d}, signature=0xb7d8908679ce6163a64e956cdec9529f6744bd14da09935e4cca17362cc07424a4db3d56b3aa21500ba33bfbb3dc1c7c185d13d9b35ed927957d6e377189bb1c2dc3f45d0afd4bcacdb344a869b7eb58d497e84284ae8264daf779d84f1d5bb3}
	at tech.pegasys.teku.core.blockvalidator.SimpleBlockValidator.verify_block_signature(SimpleBlockValidator.java:122)
	at tech.pegasys.teku.core.blockvalidator.SimpleBlockValidator.validatePreState(SimpleBlockValidator.java:72)
	at tech.pegasys.teku.core.blockvalidator.BatchBlockValidator.validatePreState(BatchBlockValidator.java:43)
	at tech.pegasys.teku.core.blockvalidator.BlockValidator.validate(BlockValidator.java:88)
	at tech.pegasys.teku.core.StateTransition.initiate(StateTransition.java:100)
	... 13 more
```

**Comment by ajsutton** [comment]
One possible cause of this is if we calculate duties while we're behind the latest head (but don't realise it for some reason).  That the shuffling will then be based on a bunch of slots being empty which later may get filled via sync but because they're all building on chain head, it's not a re-org and duties don't get recalculated.  Net result is that we wind up proposing or attesting in the wrong slots or for the wrong committees which would manifest as invalid signatures.

**Comment by ajsutton** [comment]
Fixed by #2108
